From 2325c0fedf7507f94aa3bb11cc65a70d33836f8f Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@medozas.de>
Date: Sat, 27 Aug 2011 11:12:49 +0200
Subject: [PATCH 07/19] libiptc: combine common types

Make an xt_chainlabel type out of ipt_chainlabel and ip6t_chainlabel,
and add backward-API #defines. The ABI naturally does not change
either, so no soversion bump.

Signed-off-by: Jan Engelhardt <jengelh@medozas.de>
---
 include/Makefile.am         |    2 +-
 include/libiptc/libip6tc.h  |    3 ++-
 include/libiptc/libiptc.h   |    3 ++-
 include/libiptc/xtcshared.h |    6 ++++++
 4 files changed, 11 insertions(+), 3 deletions(-)
 create mode 100644 include/libiptc/xtcshared.h

diff --git a/include/Makefile.am b/include/Makefile.am
index 0a1abea..6f7da59 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -9,4 +9,4 @@ endif
 
 nobase_include_HEADERS += \
 	libiptc/ipt_kernel_headers.h libiptc/libiptc.h \
-	libiptc/libip6tc.h libiptc/libxtc.h
+	libiptc/libip6tc.h libiptc/libxtc.h libiptc/xtcshared.h
diff --git a/include/libiptc/libip6tc.h b/include/libiptc/libip6tc.h
index 4f2d1f8..55e0bfc 100644
--- a/include/libiptc/libip6tc.h
+++ b/include/libiptc/libip6tc.h
@@ -10,10 +10,11 @@
 #	include <limits.h> /* INT_MAX in ip6_tables.h */
 #endif
 #include <linux/netfilter_ipv6/ip6_tables.h>
+#include <libiptc/xtcshared.h>
 
 struct ip6tc_handle;
 
-typedef char ip6t_chainlabel[32];
+#define ip6t_chainlabel xt_chainlabel
 
 #define IP6TC_LABEL_ACCEPT "ACCEPT"
 #define IP6TC_LABEL_DROP "DROP"
diff --git a/include/libiptc/libiptc.h b/include/libiptc/libiptc.h
index 3497d6a..ccbf6bf 100644
--- a/include/libiptc/libiptc.h
+++ b/include/libiptc/libiptc.h
@@ -10,6 +10,7 @@
 #	include <limits.h> /* INT_MAX in ip_tables.h */
 #endif
 #include <linux/netfilter_ipv4/ip_tables.h>
+#include <libiptc/xtcshared.h>
 
 #ifdef __cplusplus
 extern "C" {
@@ -17,7 +18,7 @@ extern "C" {
 
 struct iptc_handle;
 
-typedef char ipt_chainlabel[32];
+#define ipt_chainlabel xt_chainlabel
 
 #define IPTC_LABEL_ACCEPT  "ACCEPT"
 #define IPTC_LABEL_DROP    "DROP"
diff --git a/include/libiptc/xtcshared.h b/include/libiptc/xtcshared.h
new file mode 100644
index 0000000..aaf87a4
--- /dev/null
+++ b/include/libiptc/xtcshared.h
@@ -0,0 +1,6 @@
+#ifndef _LIBXTC_SHARED_H
+#define _LIBXTC_SHARED_H 1
+
+typedef char xt_chainlabel[32];
+
+#endif /* _LIBXTC_SHARED_H */
-- 
1.7.9.5

